### ISSUE TITLE
Feature request: error if scripts use both exec and trap ... EXIT

### ISSUE THREAD

**Issue by wlritchi** [koalaman/shellcheck#2631]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
cleanup() {
    echo Time to clean up!
}
trap cleanup EXIT
exec sleep 1
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

I'd like an error on the exec, noting that using exec from a script with a `trap ... EXIT` will bypass that exit trap, which is very likely not intended by the script author.

**Comment by mcandre** [comment]
Please warn on the use of any non-function-local traps, *regardless* of the particular trap signal(s) (e.g., exit, error, interruption, ...) with `exec`.

The script author is more likely to have made a mistake and assumed that the traps would persist to safeguard the control flow of the exec'ed command.
